charts/spire/charts/spire-server/templates/server-resource.yaml

```diff
@@ -79,9 +79,6 @@ metadata:
     app.kubernetes.io/component: server
 spec:
   {{- if not .Values.autoscaling.enabled }}
-  {{- if and (eq .Values.dataStore.sql.databaseType "sqlite3") (gt (int .Values.replicaCount) 1) }}
```

Comment:
It does feel like we could do with a datastore == connect flag, such that we could avoid removing this check. Both to avoid upstream drift and continue to support a SQL DB in our fork.

How about `.Values.dataStore.plugin = sql|connect`?

Reply:
We could also add a second check to ensure that a remote CA is in use?

if connect:
  fail if using disk upstream authority plugin or self-signed (none enabled) and multiple replicas
else:
  fail if using sqlite3 and multiple replicas